lang/java/ipc/pom.xml

```diff
@@ -142,6 +142,14 @@
       <artifactId>javax.annotation-api</artifactId>
       <version>1.3.2</version>
     </dependency>
+    <dependency>
```

Comment:
I think we should not define these dependencies for `avro-ipc`. We're only using this in the avro dependency itself.

Reply:
Thanks for the review and the comment @Fokko!

Hmm, I think we need to add dependencies on snappy-java and zstd-jni to the avro-ipc module, since that module contains the data interop test for Java, which reads snappy- and zstd-compressed Avro files generated by other language bindings. Without those dependencies, `./build.sh test` on the top-level directory fails as follows:

```
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M3:test (default-test) @ avro-ipc ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
Reading data files from directory: /home/sekikn/avro/lang/java/ipc/../../../build/interop/data
[INFO] Running org.apache.avro.DataFileInteropTest
Reading with specific:
Reading: py_snappy.avro
Reading with generic:
Reading: py_snappy.avro
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.349 s <<< FAILURE! - in org.apache.avro.DataFileInteropTest
[ERROR] testGeneratedSpecific(org.apache.avro.DataFileInteropTest)  Time elapsed: 0.333 s  <<< ERROR!
org.apache.avro.AvroRuntimeException: Unrecognized codec: snappy
```

In addition, I've just noticed we also have to add a dependency on zstd-jni to the avro-tools module. Due to the lack of that, the currently released avro-tools can't read zstd-compressed Avro files by itself. For this module, snappy-java is already added to the dependencies. 

```
$ curl -sLO https://www-us.apache.org/dist/avro/avro-1.9.0/java/avro-tools-1.9.0.jar
$ java -jar avro-tools-1.9.0.jar tojson build/interop/data/java_snappy.avro > /dev/null 
19/07/05 14:09:01 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
$ echo $?
0
$ java -jar avro-tools-1.9.0.jar tojson build/interop/data/java_zstandard.avro 
19/07/05 14:09:23 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Exception in thread "main" java.lang.NoClassDefFoundError: com/github/luben/zstd/ZstdOutputStream
	at org.apache.avro.file.ZstandardCodec.decompress(ZstandardCodec.java:78)
	at org.apache.avro.file.DataFileStream$DataBlock.decompressUsing(DataFileStream.java:379)
	at org.apache.avro.file.DataFileStream.hasNext(DataFileStream.java:213)
	at org.apache.avro.tool.DataFileReadTool.run(DataFileReadTool.java:80)
	at org.apache.avro.tool.Main.run(Main.java:66)
	at org.apache.avro.tool.Main.main(Main.java:55)
Caused by: java.lang.ClassNotFoundException: com.github.luben.zstd.ZstdOutputStream
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 6 more
$ echo $?
1
```